src/api_manager/context/request_context.cc

```diff
@@ -376,15 +384,30 @@ const std::string RequestContext::FindClientIPAddress() {
 void RequestContext::StartBackendSpanAndSetTraceContext() {
   backend_span_.reset(CreateSpan(cloud_trace_.get(), "Backend"));
 
-  // The span id in the header will be the backend span's id.
-  std::string trace_context = cloud_trace()->ToTraceContextHeader(
-      backend_span_->trace_span()->span_id());
+  // TODO: A better logic would be to send for GRPC backends the grpc-trace-bin
+  // header, and for http/https backends the X-Cloud-Trace-Context header.
+
+  std::string trace_context_header;
+  if (cloud_trace()->header_type() == HeaderType::GRPC_TRACE_CONTEXT) {
+    // The span id in the header will be the backend span's id.
+    std::string base64_trace_context_header =
+        cloud_trace()->ToTraceContextHeader(
+            backend_span_->trace_span()->span_id());
+    // For grpc the header must be base64 encoded because this is a binary
+    // header.
+    absl::Base64Escape(base64_trace_context_header, &trace_context_header);
```

Comment:
I prefer to have base64 encode decoding inside cloud_trace class, based on the type.

Reply:
Done.